Code of conduct (#6020)
detekt/detekt: .github/CODE_OF_CONDUCT.md

nationality, personal appearance, race, caste, color, religion, or sexual
identity and orientation.

We pledge to act and interact in ways that contribute to an open, welcoming,
diverse, inclusive, and healthy community.

## Our Standards

Examples of behavior that contributes to a positive environment for our
community include:

* Demonstrating empathy and kindness toward other people
* Being respectful of differing opinions, viewpoints, and experiences
* Giving and gracefully accepting constructive feedback
* Accepting responsibility and apologizing to those affected by our mistakes,
  and learning from the experience
* Focusing on what is best not just for us as individuals, but for the overall
  community

Examples of unacceptable behavior include:

* The use of sexualized language or imagery, and sexual attention or advances of
  any kind
* Trolling, insulting or derogatory comments, and personal or political attacks
* Public or private harassment
* Publishing others' private information, such as a physical or email address,
  without their explicit permission
* Other conduct which could reasonably be considered inappropriate in a
  professional setting

> 

Reviewer: ```suggestion
* Any form of verbal abuse on Github Issues/Pull Requests and on our public and private forums
```

Reviewer 2: I think point 2 and 3 covers this, but since these are examples, it can't hurt!

## Enforcement Responsibilities

> Community leaders are responsible for clarifying and enforcing our standards of

Reviewer: ```suggestion
Detekt maintainers are responsible for clarifying and enforcing our standards of
```

Reviewer: I would also add a link to where we can find the list of the maintainers somewhere

acceptable behavior and will take appropriate and fair corrective action in
response to any behavior that they deem inappropriate, threatening, offensive,
or harmful.

> Community leaders have the right and responsibility to remove, edit, or reject

Reviewer: Same as above

comments, commits, code, wiki edits, issues, and other contributions that are
not aligned to this Code of Conduct, and will communicate reasons for moderation
decisions when appropriate.

## Scope

This Code of Conduct applies within all community spaces, and also applies when
an individual is officially representing the community in public spaces.

> Examples of representing our community include using an official e-mail address,

Reviewer: Let's add using the Detekt logo, speaking about Detekt at conferences/podcasts/public events.

posting via an official social media account, or acting as an appointed
representative at an online or offline event.

## Enforcement

Instances of abusive, harassing, or otherwise unacceptable behavior may be
reported to the community leaders responsible for enforcement at

> security@detekt.dev.

Author: This PR is a draft because I think that we should use a different alias here so we don't mix security reports with community reports.

Reviewer 3: I created conduct@detekt.dev in our google domain.

reporter of any incident.

## Enforcement Guidelines

Community leaders will follow these Community Impact Guidelines in determining
the consequences for any action they deem in violation of this Code of Conduct:

### 1. Correction

**Community Impact**: Use of inappropriate language or other behavior deemed
unprofessional or unwelcome in the community.

**Consequence**: A private, written warning from community leaders, providing
clarity around the nature of the violation and an explanation of why the
behavior was inappropriate. A public apology may be requested.

### 2. Warning

**Community Impact**: A violation through a single incident or series of
actions.

**Consequence**: A warning with consequences for continued behavior. No
interaction with the people involved, including unsolicited interaction with
those enforcing the Code of Conduct, for a specified period of time. This
includes avoiding interactions in community spaces as well as external channels
like social media. Violating these terms may lead to a temporary or permanent
ban.

### 3. Temporary Ban

**Community Impact**: A serious violation of community standards, including
sustained inappropriate behavior.

> **Consequence**: A temporary ban from any sort of interaction or public

Reviewer: I would mention the ban from the Detekt org on Github and a report on Kotlin Slack

Reviewer 2: Are these executable?

GitHub has interaction limits feature, same possible with Slack?

Reviewer: They are but they will have to be done manually.

communication with the community for a specified period of time. No public or
private interaction with the people involved, including unsolicited interaction
with those enforcing the Code of Conduct, is allowed during this period.
Violating these terms may lead to a permanent ban.

### 4. Permanent Ban

**Community Impact**: Demonstrating a pattern of violation of community
standards, including sustained inappropriate behavior, harassment of an
individual, or aggression toward or disparagement of classes of individuals.

**Consequence**: A permanent ban from any sort of public interaction within the
community.

> ## Attribution

Reviewer 2: The last paragraph in italics of https://www.contributor-covenant.org/#how-to-adopt-contributor-covenant says we might need a badge in README.md, or is this section enough?

Author: I don't think we need a badge. We could add it later but github will show already that we had a code of conduct on the front page and link it.

Reviewer 2: I read the license and it seems to me that the attribution section at the end is enough. I'm not a lawyer, just interpreting it based on my best knowledge.

* Public or private harassment
* Publishing others' private information, such as a physical or email address,
  without their explicit permission
* Other conduct which could reasonably be considered inappropriate in a
  professional setting
* Any form of verbal abuse on Github Issues/Pull Requests and on our public and
  private forums

## Enforcement Responsibilities

[Detekt maintainers][detekt-maintainers] are responsible for clarifying and enforcing our standards of
acceptable behavior and will take appropriate and fair corrective action in
response to any behavior that they deem inappropriate, threatening, offensive,
or harmful.

[Detekt maintainers][detekt-maintainers] have the right and responsibility to remove, edit, or reject
comments, commits, code, wiki edits, issues, and other contributions that are
not aligned to this Code of Conduct, and will communicate reasons for moderation
decisions when appropriate.

## Scope

This Code of Conduct applies within all community spaces, and also applies when
an individual is officially representing the community in public spaces.
Examples of representing our community include using an official e-mail address,

> posting via an official social media account, using the detekt's logo, speaking

Reviewer 2: ```suggestion
posting via an official social media account, using detekt's logo, speaking
```
or
```suggestion
posting via an official social media account, using the detekt logo, speaking
```
